### ISSUE TITLE
Navbar Should have a responsive behaviour 

### ISSUE THREAD

**Issue by developer-diganta** [epicmaxco/vuestic-ui#4127]
**Is your feature request related to a problem? Please describe.**

Currently navbar does not show responsive behaviour i.e. if screen size decreases the navbar options decrease.

**Describe the solution you'd like**

Just like most navbars, a hamburger dropdown can be used for small screen sizes.

**Additional context**

I would like to give this issue a try if it is valid!

**Comment by m0ksem** [comment]
It would be nice to have some "template" approach or example in Navbar docs. 
If you can explain your solution in text without implementation first, I'd like to discuss it! Thanks!

fyi: Previously we were hiding left and right slots on small screen sizes, but it was an awful solution, which leads to unexpected behavior and confuse devs.

**Comment by developer-diganta** [comment]
what I was thinking was sort of controlling it more using CSS.. like have a hamburger icon (hidden for large devices) and then use the slots to display it in say a dropdown/over the entire screen sort of submenu (like some mobile navbars)? We could use device widths to trigger them?

**Comment by m0ksem** [comment]
> what I was thinking was sort of controlling it more using CSS.. like have a hamburger icon (hidden for large devices) and then use the slots to display it in say a dropdown/over the entire screen sort of submenu (like some mobile navbars)? We could use device widths to trigger them?

You can combine VaLayout and breakpoint service to achieve this. (see [Playground](https://ui.vuestic.dev/play#eNrtVm1v2zYQ/itX74NTwLbiep8MO1vTdWuGrinWYsAAAwMtnS02FEmQlBwj8H/fkZJsSk6DFRvWLxVgwOI99/bc6XgPg5daT6oSB/PBwqaGawcWXalBMLldrgbOrgZXKwnAC62MgwcwuBnBjrk0hwNsjCpgSPrDDqa0eG2Q3WnFpYth1vF0XHJCe3yqpHWwPiGXXc2L5zHO5mr3gWe4ZoaAFMbFs5PqxBY/qZ0khRau2RYb3DQc+1+I++LiOSyvIr+N8gguuP1NrbnAAHjwGhD7nVRMlN7qsxboIQcyv0hq9ogrenFYaMEcBuYWf7C3bK9KysDtBRKpOfJt7uYwvbys8oZfwrVa8J1TujkM6u9YRc7bA5+gUIbsaMMLZvarQSQSzFov2o9fnM6PxjpeBG58vK0kuLounVOB9NNTjfmGTJ7xFfv1z5wT8wSMC/UDDAuU5V9KoxzCvH4b9jV/TAVP73qqxHL02lVJ4oySDt1naaYoHZp+ojWnNwQ7crZR0o3XSmTg8N6NxRby8fTyWJ/2eXv7y23HWBJbezqwE7Yteow564NOhSjolppqXKgMRZew1SCu/veduE+6IeM5Sx2vfCvWX8lyCVNSP5ah+XborE9bZOUV0UXU9tghzA21AUhWhFZXBZLl5Cp8fp8z9ZE7EVevft543a4W0fekWg9wHmIP8EUU0Qd1RhGd/VuKdK5kzdGXU/RyTXPl/+IoOv9H3Us5diwvCsaj0dKMlbgDH5ljvpEfG2P57Iip2Dif9erw3ludxjnls1hdXwVEEyIUpb+IEHaGaY0ZcAk+WHBsO4E/VVkDMgWcsEyWTIj9BN6gQaDJDimTQEmnCEzuYS1UemeDQGJtjP4bINPUPMxxJSeLxE/4TkA/KwN4z4g9HPWs5o0jA4bqjTQAOO5GwLIMbDMSdtzl1FEV3wYH3mkYH2CFco27yNnNxtsbktnScrmF02zZUBihJpGxTMmh85ItOnAq0OSpICeECizFHhaJJ+/zdUdhsVf86LL6j4r/4lvxv37xe1egnyD1LkSvkWwwGtS747hgevLJKkn7aFi/Vo2A1tB5u5BR0Y+bpD+mW8Y5bedJUnK/y3rRJMMqaWHkLwRFlut+OqzkgXw6Sy2w4duex1QVmlY7c6t9/l3PVHu1+zWcOVPiqD1Pc0zvHjn/ZO/rEN8btGgqGvRHmWOe0lr8+sM72jkiIV3wJa2XTwl/R6tE6WOsYdelzCjsCBeivQkEUp0/2tf31PC2TcoHGtgI+MDqqydSP4U7m8yOLB7+Bs5wx/U=))

This way, dev can make any type of layout he needs. e.g. having sidebar on the right side, modal, control fixed position, etc.

**Comment by developer-diganta** [comment]
Yeah was thinking something on these lines... Should I give it a try then?

**Comment by m0ksem** [comment]
> Yeah was thinking something on these lines... Should I give it a try then?

If we can have this as ready to use templates somewhere in docs similar to layout, then I'm all in for it. We can also simply improve layout templates.
